src/iuse_actor.cpp

```diff
@@ -701,6 +702,13 @@ void consume_drug_iuse::load( const JsonObject &obj )
             effects.push_back( load_effect_data( e ) );
         }
     }
+    if( obj.has_array( "damage_over_time" ) ) {
+        for( const JsonObject e : obj.get_array( "damage_over_time" ) ) {
+            damage_over_time_data tmp_DoT_data;
+            tmp_DoT_data.load_DoT_data( e );
+            damage_over_time.emplace_back( tmp_DoT_data );
+        }
+    }
```

Comment:
```suggestion
    optional( obj, false, "damage_over_time", damage_over_time_data );
```
this works for vectors. it uses generic factory, and this is an iuse so the second parameter (was_loaded) can just be false -- this parameter is for copy-from. this is only opssible if the name of the function is `load` in the damage_over_time_data class.